lib/iris/_concatenate.py

```diff
@@ -402,6 +413,14 @@ def _coordinate_differences(self, other, attr):
                       ', '.join(diff_names))
         return result
 
+    def promote_defn(self):
+        defn = self.defn
```

Comment:
I think this is worthy of explanation !
Suggest: 
 * call it "lazy_version_defn"
 * docstring like ....
```
Produce the defn (i.e. cube.metadata) which a lazy version of the cube would have.
A cube with lazy data representing masked ints must have its `metadata.dtype` set appropriately.
```




**Even if** we think this may all change again later ...